scripts/code_coverage_report/get_diff_lines.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# This will generate a JSON file of changed files and their newly added lines.
+oIFS=$IFS
+output_file=$1
+
+json_output="["
+# Concatenate files and line indices into a JSON file.
+concatenate(){
+  local path=$1
+  shift
+  IFS=","
+  local lines=$@
+  echo "{\"file\": \"${path}\", \"added_lines\": [${lines[*]}]}"
+  IFS=$oIFS
+}
+diff-lines() {
+    local path=
+    local line=
+    local lines=()
+    while read; do
+        esc='\033'
+        # Skip lines starting with "---". e.g. "--- a/.github/workflows/database.yml".
+        if [[ "$REPLY" =~ ---\ (a/)?.* ]]; then
```

Comment:
Where does `$REPLY` from from?

Reply:
[`$REPLY`](https://tldp.org/LDP/abs/html/internalvariables.html#REPLYREF) is a default value for [`read`](https://tldp.org/LDP/abs/html/internal.html#READREF). If there are no variable assigned to `read` then each line will be assigned to `$REPLY`.